### PR TITLE
Add more useful descriptions for RLMObjectSchema, RLMSchema, and RLMProp...

### LINKED ISSUE #1500: Descriptions for RLMSchema, RLMObjectSchema, RLMProperty

### PR DESCRIPTION
**Pull request by segiddins**
...erty

Closes #1500.

\c @alazier

### CODE REVIEW

**Comment by alazier** [comment]
test?

**Comment by segiddins** [comment]
Tests added

**Comment by tgoyne** [comment]
changelog

**Comment by alazier** [comment]
Tests look fine (although quite a bit of text). Not sure why ci failed - am rerunning it.

**Comment by segiddins** [comment]
Changelog added & CI was green.

**Comment by tgoyne** [comment]
:+1:

File: Realm/RLMProperty.mm [hunk 1]
```diff
@@ -311,4 +311,8 @@ - (BOOL)isEqualToProperty:(RLMProperty *)property {
         && (_objectClassName == property->_objectClassName  || [_objectClassName isEqualToString:property->_objectClassName]);
 }
 
+- (NSString *)description {
+    return [NSString stringWithFormat:@"%@ {\n\ttype = %@\n\tobjectClassName = %@\n\tindexed = %@\n\tisPrimary = %@\n}", self.name, RLMTypeToString(self.type), self.objectClassName, self.indexed ? @"YES" : @"NO", self.isPrimary ? @"YES" : @"NO"];
```
Comment (jpsim) [inline Realm/RLMProperty.mm]: [`RLMObjectBase`'s description](https://github.com/realm/realm-cocoa/blob/067fbdbaf8dde1d725cbd1b6c8a74b7bb8c1e4bc/Realm/RLMObjectBase.mm#L225) terminates lines with a semicolon for consistency with typical descriptions in Foundation. Could you do the same here?

**Comment by jpsim** [comment]
Other than the semicolon thing, :+1:. Thanks for doing this, it's been irking me for a while.

File: Realm/RLMProperty.mm (hunk 1 above)
Comment (segiddins) [inline Realm/RLMProperty.mm]: is it just here on `RLMProperty` that you want the change?

Comment (segiddins) [inline Realm/RLMProperty.mm]: Done

**Pull request merged**